docs/Developing.md

```diff
@@ -132,14 +132,81 @@ or visit `http://localhost:3000/ui/` direct.
 
 xpcshell and mochitest
 ----------------------
-
 These are tests for the [chrome](https://developer.mozilla.org/docs/Glossary/Chrome)
 modules and files.
 
-Currently these are not runnable in the github repository, and an export to
-mozilla-central is required. We are
-[working on improving](https://bugzilla.mozilla.org/show_bug.cgi?id=1238570) this
-situation.
+**Test Export to Mozilla Central(gecko-dev)**
+
+The gecko-dev repo is cloned locally and is used as the xpcshell and 
+mochitest test build.
+
+**Resetting gecko-dev repo for pull and export**
+
+If gecko-dev has already been cloned and was used for a previous export and build,
+we can reset the repo:
+```shell
+gecko-dev$ git checkout fx-team
+gecko-dev$ git checkout -- .
+gecko-dev$ git clean -f
+gecko-dev$ git status
+``` 
+Once reset, you should not see gecko-dev/browser/extensions/loop/ directory or files.
+You can now pull fx-team latest.
+```shell
+gecko-dev$ git pull
+```  
+
+**Git export from loop repo**
+
+Export of the add-on code changes from loop to gecko-dev is done through loop makeFile.
+A loop checked out branch that is rebased to latest master can be used or master can slso be used
+for export.
+```shell
+loop$ make git-export
+gecko-dev$ git status
+```
+The add-on extension directory and files now appear in the gecko-dev repo.
+
+**Building gecko-dev**
+
+It will take some time to build.
+```shell
+gecko-dev$ ./mach clobber && ./mach build
+``` 
+
+**Running Tests**
+
+To run all the tests for both XPCShell and Mochitest 
+```shell
+gecko-dev$ ./mach test browser/extensions/loop/chrome/test
+```
+You can run all tests for XPCShell, Mochitest, specific tests and redirect output to file
+```shell
+gecko-dev$ ./mach test browser/extensions/loop/chrome/test/xpcshell/
+gecko-dev$ ./mach test browser/extensions/loop/chrome/test/xpcshell/loopservice_locales.js
+gecko-dev$ ./mach test browser/extensions/loop/chrome/test/mochitest/
+gecko-dev$ ./mach test browser/extensions/loop/chrome/test/mochitest/browser_toolbarbutton.js
+gecko-dev$ ./mach test browser/extensions/loop/chrome/test/mochitest/browser_mozLoop_infobar.js > ~/Documents/projects/infoBarTestOutput.txt
+```
+
+**Updating gecko-dev on code changes**
+
+To update gecko-dev tests from loop you must git-export again after changes have been made.
+```shell
+loop$ make git-export
+gecko-dev$ git status
+```
+If browser.ini for mochitest changes you must run this in gecko-dev then you can run tests:
+```shell
+gecko-dev$ ./mach build browser/extensions/loop
+```
+
+**Run All Tests**
+
+If you need to also run the unit tests with chrome tests.
```

Comment:
I would reword this to "Run all tests relevant to Loop" - "If you are doing big changes, you might want to run all tests that are known that could be broken by Loop. To do that, run this command:"